content/2023/asiabsdcon/fan-VT_IME_Input_Method_Editor_in_FreeBSD_vt_4.md

```diff
@@ -0,0 +1,9 @@
+---
+layout: paper
+title: "VT-IME: Input Method Editor in FreeBSD vt(4)"
+date: 2023-04-02
+author: Fan Chung
+email: fan@freebsd.org
+paper: fan-VT_IME_Input_Method_Editor_in_FreeBSD_vt_4.pdf
+---
+In this paper, we proposed vt-ime framework, which provides an environment for users to type CJK characters in the virtual terminal vt(4). The vt-ime framework is composed of two parts: the vt-ime backend and the frontend. The backend listens for key events sent from the frontend and uses an input method library, such as librime, to translate these key events into valid CJK characters. The processed characters are then sent them back to frontend for rendering. The frontend, which runs inside vt(4), is responsible for rendering the virtual terminal and display the CJK characters on the screen.
```

Comment:
The meaning of this sentence is unclear: 

> … The processed characters are then sent them back to frontend for rendering. …

Reply:
How about "The translated characters are then sent them back to the frontend for rendering."? Is it clearer now?

Reply:
I think, "them" is superfluous. 

Also, maybe avoid "back to the frontend" (because _back_ can be the opposite of _front_). 

How about: 

> The translated characters are then sent to the frontend for rendering.

Reply:
… or: 

> The translated characters are then rendered by the frontend.

(Does that change the meaning too much?)

Reply:
Sounds good